server/routes/createGoal/createGoalsController.ts

```diff
@@ -137,3 +95,96 @@ const emptyGoal = () => {
 const emptyStep = () => {
   return { title: '' }
 }
+
+const handleRemoveStep = (
+  createGoalsForm: CreateGoalsForm,
+  prisonNumber: string,
+  req: Request,
+  res: Response,
+): void => {
+  const formActionParameters = createGoalsForm.action.split('|')
+  const goalNumber = parseInt(formActionParameters[1], 10)
+  const stepNumber = parseInt(formActionParameters[2], 10)
+
+  const isValidGoalNumber =
+    !Number.isNaN(goalNumber) && goalNumber >= 0 && goalNumber <= createGoalsForm.goals.length - 1
```

Comment:
It might be slightly better to use `Number. isInteger()` here as I believe this would currently validate against decimal numbers:

```suggestion
    Number.isInteger(goalNumber) && goalNumber >= 0 && goalNumber <= createGoalsForm.goals.length - 1
```